hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaCompatibility.java

```diff
@@ -283,6 +284,35 @@ private SchemaCompatibilityResult getCompatibility(final Schema reader,
       return result;
     }
 
+    private static String getLocationName(final Deque<LocationInfo> locations, Type readerType) {
+      StringBuilder sb = new StringBuilder();
+      Iterator<LocationInfo> locationInfoIterator = locations.iterator();
+      boolean addDot = false;
+      while (locationInfoIterator.hasNext()) {
+        if (addDot) {
+          sb.append(".");
+        } else {
+          addDot = true;
+        }
+        LocationInfo next = locationInfoIterator.next();
+        sb.append(next.name);
+        //we check the reader type if we are at the last location. This is because
+        //if the type is array/map, that means the problem is that the field type
+        //of the writer is not array/map. If the type is something else, the problem
+        //is between the array element/map value of the reader and writer schemas
```

Comment:
nit: would be good to give a couple of examples that illustrate the input and expected output.